core/templates/domain/skill/SkillObjectFactory.ts

```diff
@@ -13,266 +13,292 @@
+  }
 
-    Skill.prototype.getNextMisconceptionId = function() {
-      return this._nextMisconceptionId;
-    };
+  getNextMisconceptionId(): number {
+    return this._nextMisconceptionId;
+  }
 
-    Skill.prototype.getIncrementedMisconceptionId = function(id) {
-      return id + 1;
-    };
+  getIncrementedMisconceptionId(id: string) {
+    return (parseInt(id) + 1);
+  }
 
-    Skill.prototype.getSupersedingSkillId = function() {
-      return this._supersedingSkillId;
-    };
+  getSupersedingSkillId(): string {
+    return this._supersedingSkillId;
+  }
 
-    Skill.prototype.getAllQuestionsMerged = function() {
-      return this._allQuestionsMerged;
-    };
+  getAllQuestionsMerged(): boolean {
+    return this._allQuestionsMerged;
+  }
 
-    Skill.prototype.findMisconceptionById = function(id) {
-      for (var idx in this._misconceptions) {
-        if (this._misconceptions[idx].getId() === id) {
-          return this._misconceptions[idx];
-        }
+  findMisconceptionById(id: string) {
+    var resMis;
+    this._misconceptions.forEach((mis: Misconception) => {
+      if (mis.getId() === id) {
+        resMis = mis;
       }
-      throw Error('Could not find misconception with ID: ' + id);
-    };
+    });
+    if (resMis) {
+      return resMis;
+    } else {
+      throw new Error('Cannot find Misconception by Id : ' + id);
+    }
+  }
 
-    Skill.prototype.deleteMisconception = function(id) {
-      for (var idx in this._misconceptions) {
-        if (this._misconceptions[idx].getId() === id) {
-          this._misconceptions.splice(idx, 1);
-        }
+  deleteMisconception(id: string) {
+    this._misconceptions.forEach((misc: Misconception) => {
+      if (misc.getId() === id) {
+        this._misconceptions.splice(this._misconceptions.indexOf(misc), 1);
       }
-    };
+    });
+  }
 
-    Skill.prototype.getMisconceptionAtIndex = function(idx) {
-      return this._misconceptions[idx];
-    };
+  getMisconceptionAtIndex(idx: number) {
+    return this._misconceptions[idx];
+  }
 
-    Skill.prototype.getRubricExplanation = function(difficulty) {
-      for (var idx in this._rubrics) {
-        if (this._rubrics[idx].getDifficulty() === difficulty) {
-          return this._rubrics[idx].getExplanation();
-        }
+  getRubricExplanation(difficulty: string) {
+    for (var idx in this._rubrics) {
+      if (this._rubrics[idx].getDifficulty() === difficulty) {
+        return this._rubrics[idx].getExplanation();
       }
-      return null;
-    };
+    }
+    return null;
+  }
 
-    Skill.prototype.updateRubricForDifficulty = function(
-        difficulty, explanation) {
-      if (SKILL_DIFFICULTIES.indexOf(difficulty) === -1) {
-        throw Error('Invalid difficulty value passed');
-      }
+  getMisconceptionId(index: number) {
+    return this._misconceptions[index].getId();
+  }
+
+  updateRubricForDifficulty(difficulty: string, explanation: string) {
+    if (this.SKILL_DIFFICULTIES.includes(difficulty)) {
```

Comment:
Why have you changed this line from `SKILL_DIFFICULTIES.indexOf(difficulty) === -1`, any specific reason?

Reply:
@bansalnitish  when I working on this issue I had google that thing `indexOf(difficulty) === -1`  the` indexOf` return `-1` incase of mismatching. Just a thought to make it more clearer I changed it `include` 
If you want i can change it back to `indexOf`
Thanks

Reply:
Okay, let's revert this back to original